pkg/cache/context.go

```diff
@@ -792,45 +792,54 @@ func (ctx *Context) AssumePod(name string, node string) error {
 			// assume pod volumes before assuming the pod
 			// this will update scheduler cache with essential PV/PVC binding info
 			var allBound = true
-			// volume builder might be null in UTs
-			if ctx.apiProvider.GetAPIs().VolumeBinder != nil {
-				var err error
-				// retrieve the volume claims
-				podVolumeClaims, err := ctx.apiProvider.GetAPIs().VolumeBinder.GetPodVolumeClaims(ctx.klogger, pod)
-				if err != nil {
-					log.Log(log.ShimContext).Error("Failed to get pod volume claims",
-						zap.String("podName", assumedPod.Name),
-						zap.Error(err))
-					return err
-				}
+			task := ctx.getTask(applicationID, name)
+			if task == nil {
+				log.Log(log.ShimContext).Error("BUG: task not found", zap.String("taskID", name))
+				return fmt.Errorf("task not found: %s", name)
+			}
+			task.setAllocationID(allocationID)
```

Comment:
This can be prevented if we handle the error returned in UpdateAllocation to fail the task etc.

Reply:
Done